docs/docs/behaviors/macros.md

```diff
@@ -0,0 +1,223 @@
+point in the macro bindings list, use `&macro_wait_time`, e.g. `&macro_wait_time 30`. A full example:
+
+```
+wait-ms = <10>;
+bindings
+    = <&kp F &kp A &kp S &kp T>
+    , <&macro_wait_time 500>
+    , <&kp S &kp L &kp O &kp W>
+    ;
+```
+
+### Tap Time
+
+The tap time setting controls how long a tapped behavior is held in the `bindings` list. The initial tap time for a macro, 100ms by default, can
+be set by assigning a value to the `tap-ms` property of the macro, e.g. `tap-ms = <20>;`. If you want to update the tap time at any
+point in a macro bindings list, use `&macro_tap_time`, e.g. `&macro_tap_time 30`. A full example:
+
+```
+bindings
+    = <&macro_tap_time 10>
+    , <&kp S &kp H &kp O &kp R &kp T>
+    , <&macro_tap_time 500>
+    , <&kp L &kp O &kp N &kp G>
+    ;
+```
+
+## Common Patterns
+
+### Layer Activation + More
+
+Macros make it easy to combine a [layer behavior](/docs/behaviors/layers), e.g. `&mo` with another behavior at the same time.
+Common examples are enabling one or more modifiers when the layer is active, or changing the RBG underglow color.
+
+To achieve this, a combination of a 0ms wait time and splitting the press and release between a `&macro_pause_for_release` is used:
+
+#### Layer + Modifier
+
+```
+wait-ms = <0>;
+bindings
+    = <&macro_press &mo 1 &kp LSHFT>
+    , <&macro_pause_for_release>
+    , <&macro_release &mo 1 &kp LSHFT>;
+```
+
+#### Layer + Underglow Color
+
+To trigged a different underglow when the macro is pressed, and when it is released, we use the macro "press" activation mode whenever triggering the `&rgb_ug` behavior:
+
+```
+wait-ms = <0>;
+bindings
+    = <&macro_press   &mo 1              &rgb_ug RGB_COLOR_HSB(128,100,100)>
+    , <&macro_pause_for_release>
+    , <&macro_release &mo 1 &macro_press &rgb_ug RGB_COLOR_HSB(300,100,50)>;
+```
+
+### Keycode Sequences
+
+The other common use case for macros is to sending sequences of keycodes to the connected host. Here, a wait and tap time of at least 30ms is recommended to
+avoid having HID notifications grouped at the BLE protocol level and then processed out of order:
+
+```
+wait-ms = <40>;
+tap-ms = <40>;
+bindings
+    = <&kp Z &kp M &kp K>
+    , <&kp SPACE>
+    , <&kp R &kp O &kp C &kp K &kp S>
+    ;
+```
+
+### Unicode Sequences
+
+Many operating systems allow a special sequence to input unicode characters, e.g. [Windows alt codes](https://support.microsoft.com/en-us/office/insert-ascii-or-unicode-latin-based-symbols-and-characters-d13f58d3-7bcb-44a7-a4d5-972ee12e50e0). You can use macros to insert there automatically, e.g.:
+
+```
+wait-ms = <40>;
+tap-ms = <40>;
+bindings
+    = <&macro_press   &kp LALT>
+    , <&marco_tap     &kp N0 &kp N1 &kp N6 &kp N3>
+    , <&macro_release &kp LALT>
+    ;
+```
+
+## Convenience C Macro
+
+To avoid repitition or possible typos when declaring a macro, a conenience _C_ macro, named `ZMK_MACRO(name, props)` can be used to simplify things:
```

Comment:
```suggestion
To avoid repetition or possible typos when declaring a macro, a conenience _C_ macro, named `ZMK_MACRO(name, props)` can be used to simplify things:
```